inference/core/workflows/core_steps/sinks/modbus_tcp/v1.py

```diff
@@ -0,0 +1,227 @@
+        if not response.isError():
+            print(f"Register Values: {response.registers}")
+            return response.registers
+        else:
+            print(f"Error reading registers: {response}")
+            return -1
+
+    def run(
+        self,
+        predictions: Batch[sv.Detections],
+        comm_type: str,
+        plc_ip: str,
+        plc_port: str,
+        *args,
+        **kwargs,
+    ) -> BlockResult:
+        try:
+            plc_ip = plc_ip
+            plc_port = plc_port
+
+            # Setup Modbus Client
+            modbus_client = ModbusClient(plc_ip, port=plc_port)
+
+            print("-" * 50)
+            print(f"PLC IP: {plc_ip}")
+
+            # Get prediction data
+            prediction_data = predictions.data
+            # Grab 'claas_name' from prediction_data
+            class_names = prediction_data["class_name"]
+
+            # Check if read or write
+            if comm_type.lower() == "write":
+                # TODO: HARD CODED
+                # Set PLC register value based on the label (Only supports 1 hand right now)
+                red_value = 0
+                yellow_value = 0
+                green_value = 0
+                print(f"Class Names: {class_names}")
+                for class_name in class_names:
+                    if class_name.lower() == "rock":
+                        red_value = 1
+                    elif class_name.lower() == "paper":
+                        yellow_value = 1
+                    elif class_name.lower() == "scissors":
+                        green_value = 1
+
+                print(f"Red: {red_value}")
+                print(f"Yellow: {yellow_value}")
+                print(f"Green: {green_value}")
+
+                # Write to the PLC
+                # Red
+                self.write_to_holding_register(modbus_client, 0000, red_value)
+                # Yellow
+                self.write_to_holding_register(modbus_client, 1, yellow_value)
+                # Green
+                self.write_to_holding_register(modbus_client, 2, green_value)
+
+                return_string = "Successfully wrote to PLC."
+
+            elif comm_type.lower() == "read":
+                # Read from the PLC
+                # Loop through and read all 3 registers
+                red_value = self.read_holding_registers(modbus_client, 0, 1)
+                yellow_value = self.read_holding_registers(modbus_client, 1, 1)
+                green_value = self.read_holding_registers(modbus_client, 2, 1)
+                # Print the values
+                print(f"Red: {red_value}")
+                print(f"Yellow: {yellow_value}")
+                print(f"Green: {green_value}")
+                # Concat all values into a string
+                combined_string = (
+                    f"Red: {red_value}, Yellow: {yellow_value}, Green: {green_value}"
+                )
+                return_string = combined_string
+
+            print("-" * 50)
```

Comment:
remove print

Reply:
This block is removed form this branch. It's not ready yet.